<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>L2tpWidget</class>
 <widget class="QWidget" name="L2tpWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>495</width>
    <height>422</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="labelGateway">
       <property name="text">
        <string>Gateway:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
       <property name="buddy">
        <cstring>gateway</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="gateway">
       <property name="toolTip">
        <string>L2TP server IP or name.</string>
       </property>
       <property name="clearButtonEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labelAuthType">
       <property name="text">
        <string>Authentication type:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
       <property name="buddy">
        <cstring>cmbAuthType</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="cmbAuthType">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Select an authentication mode.</string>
       </property>
       <item>
        <property name="text">
         <string>Password</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Certificates (TLS)</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <widget class="QWidget" name="page">
      <layout class="QFormLayout" name="formLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="Label_1">
         <property name="text">
          <string comment="like in Username for athentication">Username:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>username</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="username">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Set the name used for authenticating the local system to the peer to &amp;lt;name&amp;gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="Label_2">
         <property name="text">
          <string>Password:</string>
         </property>
         <property name="buddy">
          <cstring>password</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="PasswordField" name="password">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Password passed to PPPD when prompted for it.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="passwordModeEnabled" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="Label_3">
         <property name="text">
          <string>NT Domain:</string>
         </property>
         <property name="buddy">
          <cstring>domain</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="domain">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Append the domain name &amp;lt;domain&amp;gt; to the local host name for&lt;/p&gt;&lt;p&gt;authentication purposes.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <spacer name="verticalSpace">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>347</width>
           <height>200</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_1">
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="textLabel2">
         <property name="text">
          <string>CA Certificate:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>userCA</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="KUrlRequester" name="userCA">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Certificate authority (CA) file in .pem, .der, .crt, .cer or .p12 formats.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.der *.crt *.cer *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="textLabel3_2">
         <property name="text">
          <string>User Certificate:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>userCert</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="KUrlRequester" name="userCert">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Certificate in .pem, .der, .crt, .cer or .p12 formats.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.der *.crt *.cer *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="textLabel4">
         <property name="text">
          <string>Private Key:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>userKey</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="KUrlRequester" name="userKey">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Private key in .pem, .der, .key, .pk8 or .p12 formats.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.der *.key *.pk8 *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Private Key Password:</string>
         </property>
         <property name="buddy">
          <cstring>userKeyPassword</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="PasswordField" name="userKeyPassword">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Password for private key or PKCS#12 certificate.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="passwordModeEnabled" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnIPSecSettings">
       <property name="text">
        <string>IPsec Settings…</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnPPPSettings">
       <property name="text">
        <string>PPP Settings…</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QWidget</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>PasswordField</class>
   <extends>QLineEdit</extends>
   <header>passwordfield.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>gateway</tabstop>
  <tabstop>cmbAuthType</tabstop>
  <tabstop>username</tabstop>
  <tabstop>password</tabstop>
  <tabstop>domain</tabstop>
  <tabstop>userCA</tabstop>
  <tabstop>userCert</tabstop>
  <tabstop>userKey</tabstop>
  <tabstop>userKeyPassword</tabstop>
  <tabstop>btnIPSecSettings</tabstop>
  <tabstop>btnPPPSettings</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>cmbAuthType</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>345</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
